test/libdnf/base_test_case.cpp

```diff
@@ -42,12 +42,12 @@ using fmt::format;
 
 libdnf::repo::RepoWeakPtr BaseTestCase::add_repo(const std::string & repoid, const std::string & repo_path, bool load) {
     auto repo = repo_sack->create_repo(repoid);
-
     repo->get_config().baseurl().set("file://" + repo_path);
 
     if (load) {
-        repo->fetch_metadata();
-        repo->load();
+        libdnf::repo::RepoQuery repos(base);
+        repos.filter_id(repoid);
+        repo_sack->update_and_load_repos(repos);
```

Comment:
I have a bad feeling about possibility to load subset of repositories using `update_and_load_repos(repos)`. This is not a problem of the PR but our API. The reason is modularity, where it is strongly recommended to have one point of application of modular filtering. The PR is going into right directions.